core/src/main/java/org/apache/spark/memory/MemoryMode.java

```diff
@@ -22,5 +22,6 @@
 @Private
 public enum MemoryMode {
   ON_HEAP,
-  OFF_HEAP
+  OFF_HEAP,
+  ON_HEAP_CACHEDBATCH
```

Comment:
we can make `ColumnVector.allocate` accept a `VectorType`(not exists yet) instead of `MemoryMode`